server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1114,6 +1138,55 @@ private DefaultSearchContext createSearchContext(ReaderContext reader, ShardSear
         return searchContext;
     }
 
+    private void writeTaskResourceUsage(SearchShardTask task) {
+        try {
+            // Get resource usages when task starts
+            ThreadResourceInfo threadResourceInfo = task.getActiveThreadResourceInfo(
+                Thread.currentThread().getId(),
+                ResourceStatsType.WORKER_STATS
+            );
+            if (threadResourceInfo == null) {
+                return;
+            }
+            Map<ResourceStats, ResourceUsageInfo.ResourceStatsInfo> startValues = threadResourceInfo.getResourceUsageInfo().getStatsInfo();
+            if (startValues.containsKey(ResourceStats.CPU) && startValues.containsKey(ResourceStats.MEMORY)) {
```

Comment:
Why are we exiting early here ? If we want to capture the resource Usage for this task then shouldn't we go ahead irrespective of this predicate.

Till this line in this method all this means in plain words is that `Task has been scheduled on a thread at least once irrespective of its current state`.

Reply:
Good catch :P, fixed this bug in the latest version.